wlauto/workloads/stress_ng/__init__.py

```diff
@@ -0,0 +1,96 @@
+                                  'yield', 'hdd', 'cache', 'sock', 'fallocate',
+                                  'flock', 'affinity', 'timer', 'dentry',
+                                  'urandom', 'sem', 'open', 'sigq', 'poll'],
+                  description='Stress test case name. The cases listed in '
+                              'allowed values come from the stable release '
+                              'version 0.01.32. The binary included here '
+                              'compiled from dev version 0.06.01. Refer to '
+                              'man page for the definition of each stressor.'),
+        Parameter('threads', kind=int, default=0,
+                  description='The number of workers to run. Specifying a '
+                              'negative or zero value will select the number '
+                              'of online processors.'),
+        Parameter('duration', kind=int, default=60,
+                  description='Timeout for test execution in seconds')
+    ]
+
+    def setup(self, context):
+        host_binary = context.resolver.get(Executable(self, self.device.abi,
+                                                      'stress-ng'))
+        self.binary = self.device.install_if_needed(host_binary)
+        self.log = os.path.join(self.device.working_directory,
+                                'stress_ng_output.txt')
+        self.results = os.path.join(self.device.working_directory,
+                                    'stress_ng_results.yaml')
+        self.command = ('{} --{} {} --timeout {}s --log-file {} --yaml {} '
+                        '--metrics-brief --verbose'
+                        .format(self.binary, self.stressor, self.threads,
+                                self.duration, self.log, self.results))
+        self.timeout = self.duration + 10
+
+    def run(self, context):
+        self.output = self.device.execute(self.command, timeout=self.timeout,
+                                          as_root=True)
+
+    def update_result(self, context):
+        host_file_log = os.path.join(context.output_directory,
+                                     'stress_ng_output.txt')
+        host_file_results = os.path.join(context.output_directory,
+                                         'stress_ng_results.yaml')
+        self.device.pull_file(self.log, host_file_log)
+        self.device.pull_file(self.results, host_file_results)
+
+        with open(host_file_results, 'r') as stress_ng_results:
+            results = yaml.load(stress_ng_results)
+
+        try:
+            metric = results['metrics'][0]['stressor']
+            throughput = results['metrics'][0]['bogo-ops']
+            context.result.add_metric(metric, throughput, 'ops')
+        # For some stressors like vm, if test duration is too short, stress_ng
+        # may not able to produce test throughput rate.
+        except TypeError:
+            pass
```

Comment:
It's probably worth logging a warning here stating that a result could not be obtained because the test duration was set too short. Better yet, if you know which stressors require longer tests, it's worth adding a `validate()` method where you check that the duration specified is compatible with the specified stressor.